packages/api/src/controllers/animalController.js

```diff
@@ -0,0 +1,78 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import { Model, transaction } from 'objection';
+import AnimalModel from '../models/animalModel.js';
+import baseController from './baseController.js';
+
+const animalController = {
+  getFarmAnimals() {
+    return async (req, res) => {
+      try {
+        const { farm_id } = req.headers;
+        const rows = await AnimalModel.query().where({ farm_id });
```

Comment:
Are we waiting to add query parameters?

Reply:
I don't think we'll need query options for this one for now, the use case I could think of for query options in this scenario would be filtering, but since we need to pre-fetch everything upfront anyway for offline purposes we'll end up doing the filtering on the client side. For creation there's no offline support so we can fetch only breeds for the type they've selected in the form. That being said, we can add query options for completeness if we want to have that tool at hand in case we end up needing it.

Reply:
Whatever you think is best! Just something I noticed.